lib/enumerable/each_with_index.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    input = is_a?(Range) ? to_a : self
+    my_each { |index| yield(input[index], index) }
+    input
+  end
```

Comment:
There is an error here. See the last results
```ruby
irb(main):010:0> each_output = ''
=> ""
irb(main):011:0> block = proc { |num, idx| each_output += "Num: #{num}, idx: #{idx}\n" }
=> #<Proc:0x0000559e2d845af0@(irb):11>
irb(main):012:0> array.each_with_index(&block)
=> [5, 1, 7, 2, 3, 6, 3, 3, 1, 6, 1, 2, 5, 8, 3, 7, 2, 6, 5, 7, 5, 4, 3, 6, 4, 0, 8, 6, 3, 8, 0, 3, 7, 1, 0, 4, 0, 8, 8, 6, 1, 3, 0, 7, 0, 5, 1, 0, 2, 2, 8, 1, 3, 3, 1, 6, 0, 2, 8, 2, 8, 7, 3, 7, 3, 3, 0, 8, 6, 5, 5, 1, 6, 0, 3, 4, 1, 1, 7, 1, 5, 8, 5, 0, 7, 5, 0, 6, 4, 0, 3, 1, 6, 7, 6, 6, 7, 7, 4, 7]
irb(main):013:0> my_each_output = ''
=> ""
irb(main):014:0> block = proc { |num, idx| my_each_output += "Num: #{num}, idx: #{idx}\n" }
=> #<Proc:0x0000559e2d9a5378@(irb):14>
irb(main):015:0> array.my_each_with_index(&block)
=> [5, 1, 7, 2, 3, 6, 3, 3, 1, 6, 1, 2, 5, 8, 3, 7, 2, 6, 5, 7, 5, 4, 3, 6, 4, 0, 8, 6, 3, 8, 0, 3, 7, 1, 0, 4, 0, 8, 8, 6, 1, 3, 0, 7, 0, 5, 1, 0, 2, 2, 8, 1, 3, 3, 1, 6, 0, 2, 8, 2, 8, 7, 3, 7, 3, 3, 0, 8, 6, 5, 5, 1, 6, 0, 3, 4, 1, 1, 7, 1, 5, 8, 5, 0, 7, 5, 0, 6, 4, 0, 3, 1, 6, 7, 6, 6, 7, 7, 4, 7]
irb(main):016:0> each_output
=> "Num: 5, idx: 0\nNum: 1, idx: 1\nNum: 7, idx: 2\nNum: 2, idx: 3\nNum: 3, idx: 4\nNum: 6, idx: 5\nNum: 3, idx: 6\nNum: 3, idx: 7\nNum: 1, idx: 8\nNum: 6, idx: 9\nNum: 1, idx: 10\nNum: 2, idx: 11\nNum: 5, idx: 12\nNum: 8, idx: 13\nNum: 3, idx: 14\nNum: 7, idx: 15\nNum: 2, idx: 16\nNum: 6, idx: 17\nNum: 5, idx: 18\nNum: 7, idx: 19\nNum: 5, idx: 20\nNum: 4, idx: 21\nNum: 3, idx: 22\nNum: 6, idx: 23\nNum: 4, idx: 24\nNum: 0, idx: 25\nNum: 8, idx: 26\nNum: 6, idx: 27\nNum: 3, idx: 28\nNum: 8, idx: 29\nNum: 0, idx: 30\nNum: 3, idx: 31\nNum: 7, idx: 32\nNum: 1, idx: 33\nNum: 0, idx: 34\nNum: 4, idx: 35\nNum: 0, idx: 36\nNum: 8, idx: 37\nNum: 8, idx: 38\nNum: 6, idx: 39\nNum: 1, idx: 40\nNum: 3, idx: 41\nNum: 0, idx: 42\nNum: 7, idx: 43\nNum: 0, idx: 44\nNum: 5, idx: 45\nNum: 1, idx: 46\nNum: 0, idx: 47\nNum: 2, idx: 48\nNum: 2, idx: 49\nNum: 8, idx: 50\nNum: 1, idx: 51\nNum: 3, idx: 52\nNum: 3, idx: 53\nNum: 1, idx: 54\nNum: 6, idx: 55\nNum: 0, idx: 56\nNum: 2, idx: 57\nNum: 8, idx: 58\nNum: 2, idx: 59\nNum: 8, idx: 60\nNum: 7, idx: 61\nNum: 3, idx: 62\nNum: 7, idx: 63\nNum: 3, idx: 64\nNum: 3, idx: 65\nNum: 0, idx: 66\nNum: 8, idx: 67\nNum: 6, idx: 68\nNum: 5, idx: 69\nNum: 5, idx: 70\nNum: 1, idx: 71\nNum: 6, idx: 72\nNum: 0, idx: 73\nNum: 3, idx: 74\nNum: 4, idx: 75\nNum: 1, idx: 76\nNum: 1, idx: 77\nNum: 7, idx: 78\nNum: 1, idx: 79\nNum: 5, idx: 80\nNum: 8, idx: 81\nNum: 5, idx: 82\nNum: 0, idx: 83\nNum: 7, idx: 84\nNum: 5, idx: 85\nNum: 0, idx: 86\nNum: 6, idx: 87\nNum: 4, idx: 88\nNum: 0, idx: 89\nNum: 3, idx: 90\nNum: 1, idx: 91\nNum: 6, idx: 92\nNum: 7, idx: 93\nNum: 6, idx: 94\nNum: 6, idx: 95\nNum: 7, idx: 96\nNum: 7, idx: 97\nNum: 4, idx: 98\nNum: 7, idx: 99\n"
irb(main):017:0> my_each_output
=> "Num: 6, idx: 5\nNum: 1, idx: 1\nNum: 3, idx: 7\nNum: 7, idx: 2\nNum: 2, idx: 3\nNum: 3, idx: 6\nNum: 2, idx: 3\nNum: 2, idx: 3\nNum: 1, idx: 1\nNum: 3, idx: 6\nNum: 1, idx: 1\nNum: 7, idx: 2\nNum: 6, idx: 5\nNum: 1, idx: 8\nNum: 2, idx: 3\nNum: 3, idx: 7\nNum: 7, idx: 2\nNum: 3, idx: 6\nNum: 6, idx: 5\nNum: 3, idx: 7\nNum: 6, idx: 5\nNum: 3, idx: 4\nNum: 2, idx: 3\nNum: 3, idx: 6\nNum: 3, idx: 4\nNum: 5, idx: 0\nNum: 1, idx: 8\nNum: 3, idx: 6\nNum: 2, idx: 3\nNum: 1, idx: 8\nNum: 5, idx: 0\nNum: 2, idx: 3\nNum: 3, idx: 7\nNum: 1, idx: 1\nNum: 5, idx: 0\nNum: 3, idx: 4\nNum: 5, idx: 0\nNum: 1, idx: 8\nNum: 1, idx: 8\nNum: 3, idx: 6\nNum: 1, idx: 1\nNum: 2, idx: 3\nNum: 5, idx: 0\nNum: 3, idx: 7\nNum: 5, idx: 0\nNum: 6, idx: 5\nNum: 1, idx: 1\nNum: 5, idx: 0\nNum: 7, idx: 2\nNum: 7, idx: 2\nNum: 1, idx: 8\nNum: 1, idx: 1\nNum: 2, idx: 3\nNum: 2, idx: 3\nNum: 1, idx: 1\nNum: 3, idx: 6\nNum: 5, idx: 0\nNum: 7, idx: 2\nNum: 1, idx: 8\nNum: 7, idx: 2\nNum: 1, idx: 8\nNum: 3, idx: 7\nNum: 2, idx: 3\nNum: 3, idx: 7\nNum: 2, idx: 3\nNum: 2, idx: 3\nNum: 5, idx: 0\nNum: 1, idx: 8\nNum: 3, idx: 6\nNum: 6, idx: 5\nNum: 6, idx: 5\nNum: 1, idx: 1\nNum: 3, idx: 6\nNum: 5, idx: 0\nNum: 2, idx: 3\nNum: 3, idx: 4\nNum: 1, idx: 1\nNum: 1, idx: 1\nNum: 3, idx: 7\nNum: 1, idx: 1\nNum: 6, idx: 5\nNum: 1, idx: 8\nNum: 6, idx: 5\nNum: 5, idx: 0\nNum: 3, idx: 7\nNum: 6, idx: 5\nNum: 5, idx: 0\nNum: 3, idx: 6\nNum: 3, idx: 4\nNum: 5, idx: 0\nNum: 2, idx: 3\nNum: 1, idx: 1\nNum: 3, idx: 6\nNum: 3, idx: 7\nNum: 3, idx: 6\nNum: 3, idx: 6\nNum: 3, idx: 7\nNum: 3, idx: 7\nNum: 3, idx: 4\nNum: 3, idx: 7\n"
irb(main):018:0> array
=> [5, 1, 7, 2, 3, 6, 3, 3, 1, 6, 1, 2, 5, 8, 3, 7, 2, 6, 5, 7, 5, 4, 3, 6, 4, 0, 8, 6, 3, 8, 0, 3, 7, 1, 0, 4, 0, 8, 8, 6, 1, 3, 0, 7, 0, 5, 1, 0, 2, 2, 8, 1, 3, 3, 1, 6, 0, 2, 8, 2, 8, 7, 3, 7, 3, 3, 0, 8, 6, 5, 5, 1, 6, 0, 3, 4, 1, 1, 7, 1, 5, 8, 5, 0, 7, 5, 0, 6, 4, 0, 3, 1, 6, 7, 6, 6, 7, 7, 4, 7]
irb(main):019:0> 

```

Reply:
Seemed like the problem was reusing `#my_each`. Solved